lib/_stream_duplex.js

```diff
@@ -70,63 +70,60 @@ function Duplex(options) {
   }
 }
 
-ObjectDefineProperty(Duplex.prototype, 'writableHighWaterMark', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get() {
-    return this._writableState && this._writableState.highWaterMark;
-  }
-});
+ObjectDefineProperties(Duplex.prototype, {
 
-ObjectDefineProperty(Duplex.prototype, 'writableBuffer', {
-  // Making it explicit this property is not enumerable
-  // because otherwise some prototype manipulation in
-  // userland will fail
-  enumerable: false,
-  get: function() {
-    return this._writableState && this._writableState.getBuffer();
-  }
-});
+  destroyed: {
+    get() {
+      if (this._readableState === undefined ||
+        this._writableState === undefined) {
+        return false;
+      }
+      return this._readableState.destroyed && this._writableState.destroyed;
```

Comment:
Nit: this could be simplified to:

```js
return this._readableState &&
       this._readableState.destroyed &&
       this._writableState &&
       this._writableState.destroyed;
```

Reply:
This would return `undefined` instead of `false` if `_readableState` is `undefined`.

Reply:
Good point. It could be wrapped into `Boolean(condition)` to have a single expression but it's fine as it is.